Feature/tf assignment

TaeAhmed opened the pull request:
task:
1-create two modules one for vpc
2-in the vpc create 3 public and 3 private subnets 
3-the 2nd module for ec2 
4-create two ec2s one for web and the other for database 
5-the web server depends on the database server
6-the database server should not be deletable
7-seperate the variables and outputs and locals from the resources files

terraform plan output is in out.txt
could not complete 6 due to lifecycle prevent_destroy in module not allowing variables, 
however I may accomplish this by defining it's resource in main.tf instead of using the ec2 module

TaeAhmed committed:
snapshot 01

TaeAhmed committed:
completed 1,2

TaeAhmed committed:
completed 1,2

TaeAhmed committed:
completed all except 6

TaeAhmed committed:
added a variable

MoustafaRoushdy reviewed (approved):
الكود مش كامل طبعا بس لحد دلوقتي صح

MoustafaRoushdy commented on main.tf:
```diff
@@ -0,0 +1,92 @@
+    subnet_id = aws_subnet.public[0].id
+}
+
+resource "aws_eip" "nat_eip" {
+    vpc = true
+    tags = { Name = "nat_eip"}
+}
+
+resource "aws_route_table" "public" {
+    vpc_id = module.vpc.vpc_id
+}
+
+resource "aws_route_table" "private" {
+    vpc_id = module.vpc.vpc_id
+}
+
+resource "aws_route" "public_route" {
+    route_table_id = aws_route_table.public.id
+    destination_cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.igw.id
+}
+
+resource "aws_route" "private_route" {
+    route_table_id = aws_route_table.private.id
+    destination_cidr_block = "0.0.0.0/0"
+    nat_gateway_id = aws_nat_gateway.natgw.id
+}
+
+resource "aws_route_table_association" "public" {
+    count = 3
+    subnet_id = aws_subnet.public[count.index].id
+    route_table_id = aws_route_table.public.id
+}
+
+resource "aws_route_table_association" "private" {
+    count = 3
+    subnet_id = aws_subnet.private[count.index].id
+    route_table_id = aws_route_table.private.id
+}
+
```
should be in the vpc module

TaeAhmed replied:
done

MoustafaRoushdy commented on modules/ec2/main.tf:
```diff
@@ -0,0 +1,12 @@
+    lifecycle {
+        prevent_destroy = false
+        #prevent_destroy = var.prevent_destroy
+        #2<< variables may not be used here
```
?

TaeAhmed replied:
In the new commit I Separated modules for web and db ec2;
    reason: prevent_destroy doesn't accept configuration through variables

TaeAhmed committed:
- Expanded the vpc module
- Seperate module for web and db ec2:
	reason: prevent_destroy doesn't accept configuration through variables
